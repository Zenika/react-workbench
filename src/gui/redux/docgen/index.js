export { default } from './docgen'
export * from './docgen.selectors'
export * from './docgen.api'
