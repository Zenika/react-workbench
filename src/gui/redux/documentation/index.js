export { default } from './documentation'
export * from './documentation.api'
