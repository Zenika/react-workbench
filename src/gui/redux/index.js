export * from './store'
export { default } from './store'
