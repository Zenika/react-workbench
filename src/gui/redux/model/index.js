export { default } from './model'
export * from './model.selectors'
export * from './model.actions'
