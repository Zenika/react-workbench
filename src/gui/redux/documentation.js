import { combineReducers } from 'redux'
import { simpleObject } from 'trampss-redux-factory'

const options = { path: 'documentation' }
export const readme = simpleObject({ ...options, name: 'readme' })
export const html = simpleObject({ ...options, name: 'html' })

export default combineReducers({
  readme,
  html,
})
