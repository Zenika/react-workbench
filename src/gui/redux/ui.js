import { combineReducers } from 'redux'
import { simpleObject } from 'trampss-redux-factory'

const options = { path: 'ui', prefix: 'ui' }
export const state = simpleObject({ ...options, defaultData: '', name: 'state' })

export default combineReducers({
  state,
})
