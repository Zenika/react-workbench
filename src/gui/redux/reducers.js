import { combineReducers } from 'redux'
import { simpleObject, keyValue } from 'trampss-redux-factory'
import component from './component'
import documentation from './documentation'
import router from './router'
import ui from './ui'

export const docgen = simpleObject({ defaultData: { props: {} }, name: 'docgen' })
export const states = keyValue({ key: 'name', name: 'states' })
export const unhandledErrors = keyValue({ key: 'timestamp', name: 'unhandledErrors' })

export default combineReducers({
  component,
  docgen,
  documentation,
  router: router.reducer,
  states,
  ui,
  unhandledErrors,
})
