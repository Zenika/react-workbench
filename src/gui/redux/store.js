import { createStore, applyMiddleware } from 'redux'
import { composeWithDevTools } from 'redux-devtools-extension'
import { initializeCurrentLocation } from 'redux-little-router'
import createSagaMiddleware from 'redux-saga'
import thunk from 'redux-thunk'
import sagas from '../sagas'
import router from './router'
import reducers from './reducers'

const sagaMiddleware = createSagaMiddleware()

// initialize redux store
const store = createStore(
  reducers,
  composeWithDevTools(router.enhancer, applyMiddleware(thunk, sagaMiddleware, router.middleware))
)

sagaMiddleware.run(sagas(store))

// initialize routing
const initialLocation = store.getState().router
if (initialLocation) {
  store.dispatch(initializeCurrentLocation(initialLocation))
}

export default store
