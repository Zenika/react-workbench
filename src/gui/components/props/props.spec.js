/* eslint-disable react/jsx-filename-extension */
/* eslint-env jest */
import React from 'react'
import renderer from 'react-test-renderer'
import { createStore } from 'redux'
import { Provider } from 'react-redux'
import snap from 'misc/snap'
import Props from './props'
import PropsContainer from './props.container'

// mock input component into props
jest.mock('./input', () => 'input')

describe('component/props', () => {
  describe('props.jsx', () => {
    const snapshot = snap(Props)
    it('should take custom className', snapshot({ className: 'custom' }))
    it('should take custom style', snapshot({ style: { backgroundColor: 'red' } }))
    it('should render by default', snapshot({ }))
    it('should render inputs following given properties', snapshot({ properties: ['prop1', 'prop2'] }))
  })

  describe('props.container.js', () => {
    it('should map redux model into properties array', () => {
      const store = createStore(() => ({ model: { foo: { type: 'string' }, bar: { type: 'string' } } }))
      const container = renderer.create(
        <Provider store={store}>
          <PropsContainer />
        </Provider>
      )
      const tree = container.toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
