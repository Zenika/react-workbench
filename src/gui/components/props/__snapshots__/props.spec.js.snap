// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component/props props.container.js should map redux model into properties array 1`] = `
<div
  className="props"
  style={Object {}}
>
  <input
    name="foo"
  />
  <input
    name="bar"
  />
</div>
`;

exports[`component/props props.jsx should render by default 1`] = `
<div
  className="props"
  style={Object {}}
/>
`;

exports[`component/props props.jsx should render inputs following given properties 1`] = `
<div
  className="props"
  style={Object {}}
>
  <input
    name="prop1"
  />
  <input
    name="prop2"
  />
</div>
`;

exports[`component/props props.jsx should take custom className 1`] = `
<div
  className="props custom"
  style={Object {}}
/>
`;

exports[`component/props props.jsx should take custom style 1`] = `
<div
  className="props"
  style={
    Object {
      "backgroundColor": "red",
    }
  }
/>
`;
